### DISCUSSION TITLE
Standalone AVRC for Playback Control

### DISCUSSION THREAD

**Discussion by rav345** [pschatzmann/arduino-audio-tools#831] [Q&A] [answered]
Hi,
I am trying to use an ESP32/Arduino BLE/RP2040 as a remote controller for issuing play/pause/volume_control commands to a smartphone or directly to a headset streaming from a smartphone. I see that AVRCP is tightly linked with the A2DP. When I use ESP32 WROOM32 as a BluetoothA2DPSink and issue play/pause commands while headphones are also paired, the music starts to stream to ESP32. 
What is the best way to use AVRCP as a standalone system? I want to do this to issue commands based on the sensor inputs. Please suggest :-)

**Comment by pschatzmann** [comment, marked as answer]
Not sure if this is possible at all. The best way to deal with this is to learn the AVRC API of the EP32: https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/bluetooth/esp_avrc.html

**Reply by rav345** [reply]
Thank you!
I suppose it is also not possible to use one ESP32 as BluetoothA2DP sink while pairing BT headphones to stream the music to them, or is it? Any other design ideas, perhaps?

**Reply by pschatzmann** [reply]
It is not possible with my library. Therefore I gave you the link to the core functionality.
Are you sure that the headphones support this ?

**Reply by rav345** [reply]
The way I tested it: Sony XM4 earbuds paired to MacBook Pro + BluetoothA2DPSink on ESP32 paired to Macbook Pro. The sink was set up to toggle between a2dp_sink.play()/pause() every 10s. It worked while I streamed from youtube via the web browser. However, for some reason, when I do the same with an iPhone, the music stream is diverted to the sink after the first issue of the pause/play command.

> I suppose it is also not possible to use one ESP32 as BluetoothA2DP sink while pairing BT headphones to stream the music to them, or is it? Any other design ideas, perhaps?

In this case, I wondered if just one ESP32 can act as a router of audio stream from the smartphone while also controlling the playback as a sink. Or how about two?
